api/src/main/java/org/pmiops/workbench/institution/InstitutionServiceImpl.java

```diff
@@ -128,20 +138,25 @@ public void deleteInstitution(final String shortName) {
   @Override
   public Optional<Institution> updateInstitution(
       final String shortName, final Institution institutionToUpdate) {
-    return institutionDao
-        .findOneByShortName(shortName)
-        .map(DbInstitution::getInstitutionId)
-        .map(
-            dbId -> {
-              // create new DB object, but mark it with the original's ID to indicate that this is
-              // an update
-
-              final DbInstitution dbObjectToUpdate =
-                  institutionDao.save(
-                      institutionMapper.modelToDb(institutionToUpdate).setInstitutionId(dbId));
-              populateAuxTables(institutionToUpdate, dbObjectToUpdate);
-              return toModel(dbObjectToUpdate);
-            });
+    validateInstitution(institutionToUpdate);
+    try {
+      return institutionDao
+          .findOneByShortName(shortName)
+          .map(DbInstitution::getInstitutionId)
+          .map(
+              dbId -> {
+                // create new DB object, but mark it with the original's ID to indicate that this is
+                // an update
+
+                final DbInstitution dbObjectToUpdate =
+                    institutionDao.save(
+                        institutionMapper.modelToDb(institutionToUpdate).setInstitutionId(dbId));
+                populateAuxTables(institutionToUpdate, dbObjectToUpdate);
+                return toModel(dbObjectToUpdate);
+              });
+    } catch (DataIntegrityViolationException ex) {
+      throw new ConflictException("Display Name already exist");
```

Comment:
The assumption is that request will be coming from UI ,the shortName will be different as we will be generating it  and just display Name is something we will not have control over and can be same.

Reply:
Conflict Exception throws 409 exception that i can use in UI to show appropriate message

Reply:
Good point.  ConflictException is good, let's use that.  But we should include the original exception message in the text.  Maybe something like:
`new ConflictException("DataIntegrityViolationException: " + ex.getMessage())`